src/messages/text_document_completion.cc

```diff
@@ -296,6 +296,14 @@ struct TextDocumentCompletionHandler : MessageHandler {
             out.result.items = results;
 
             // Emit completion results.
+            if (existing_completion.size()==0 && is_global_completion) {
+              LOG_S(INFO) << "Existing completion is empty, no completion results will be returned";
```

Comment:
Nit: It's better to check `existing_completion.empty()` because, for some containers, `size()` is not guaranteed to be a constant-time operation.

Reply:
Nit: We can move lines 294-296 after this `if` block so that, when this `if` is true, we won't create and immediately destroy an unused `Out_TextDocumentComplete` structure.